it/it-tests/src/test/java/it/settings/EmailsTest.java

```diff
@@ -0,0 +1,147 @@
+
+  @BeforeClass
+  public static void before() throws Exception {
+    ADMIN_WS_CLIENT = newAdminWsClient(orchestrator);
+    SETTINGS = ADMIN_WS_CLIENT.settingsService();
+
+    SMTP_SERVER = new Wiser(0);
+    SMTP_SERVER.start();
+    System.out.println("SMTP Server port: " + SMTP_SERVER.getServer().getPort());
+  }
+
+  @AfterClass
+  public static void stop() {
+    if (SMTP_SERVER != null) {
+      SMTP_SERVER.stop();
+    }
+    setServerProperty(orchestrator, "email.smtp_host.secured", null);
+    setServerProperty(orchestrator, "email.smtp_port.secured", null);
+    setServerProperty(orchestrator, "email.smtp_secure_connection.secured", null);
+    setServerProperty(orchestrator, "email.smtp_username.secured", null);
+    setServerProperty(orchestrator, "email.smtp_password.secured", null);
+    setServerProperty(orchestrator, "email.from", null);
+    setServerProperty(orchestrator, "email.prefix", null);
+  }
+
+  @Before
+  public void prepare() {
+    orchestrator.resetData();
+    SMTP_SERVER.getMessages().clear();
+  }
+
+  @Test
+  public void update_email_settings() throws Exception {
+    updateEmailSettings("localhost", "42", "noreply@email.com", "[EMAIL]", "ssl", "john", "123456");
+
+    Settings.ValuesWsResponse response = SETTINGS.values(ValuesRequest.builder()
+      .setKeys("email.smtp_host.secured", "email.smtp_port.secured", "email.smtp_secure_connection.secured", "email.smtp_username.secured", "email.smtp_password.secured",
+        "email.from", "email.prefix")
+      .build());
+
+    assertThat(response.getSettingsList()).extracting("key", "value")
+      .containsOnly(
+        tuple("email.smtp_host.secured", "localhost"),
+        tuple("email.smtp_port.secured", "42"),
+        tuple("email.smtp_secure_connection.secured", "ssl"),
+        tuple("email.smtp_username.secured", "john"),
+        tuple("email.smtp_password.secured", "123456"),
+        tuple("email.from", "noreply@email.com"),
+        tuple("email.prefix", "[EMAIL]"));
+  }
+
+  @Test
+  public void send_test_email() throws Exception {
+    updateEmailSettings("localhost", Integer.toString(SMTP_SERVER.getServer().getPort()), null, null, null, null, null);
+
+    sendTestEmail("test@example.org", "Test Message from SonarQube", "This is a test message from SonarQube");
+
+    // We need to wait until all notifications will be delivered
+    waitUntilAllNotificationsAreDelivered();
+    Iterator<WiserMessage> emails = SMTP_SERVER.getMessages().iterator();
+    MimeMessage message = emails.next().getMimeMessage();
+    assertThat(message.getHeader("To", null)).isEqualTo("<test@example.org>");
+    assertThat(message.getSubject()).contains("Test Message from SonarQube");
+    assertThat((String) message.getContent()).contains("This is a test message from SonarQube");
+    assertThat(emails.hasNext()).isFalse();
+  }
+
+  private static void waitUntilAllNotificationsAreDelivered() throws InterruptedException {
+    Thread.sleep(10000);
```

Comment:
It's like that in other ITs.... I completely agree we should find a better way, but I haven't idea for the moment.